backend/reg.ml

```diff
@@ -79,6 +87,14 @@ let create_gen ~name ~typ ~loc =
 
 let create typ = create_gen ~name:Name.Anon ~typ ~loc:Unknown
 
+let create_alias t ~typ =
+  match t.loc with
+  | Unknown ->
+      Misc.fatal_errorf "Reg.create_alias for unknown register %s"
+       (Name.to_string t.name)
+  | Reg _ | Stack _ ->
```

Comment:
Given the associated docstring,
I wonder whether we should
also error out on `Stack _`?